ukis_pysat/raster.py

```diff
@@ -284,7 +292,40 @@ def dn2toa(self, platform, mtl_file=None, mtd_file=None, wavelengths=None):
             else:
                 # get rescale factors from mtl file
                 mtl = toa_utils._load_mtl(str(mtl_file))  # no obvious reason not to call this
-                metadata = mtl["L1_METADATA_FILE"]
+
+                # search for collection number in MTL metadata file
+                def iterate_mtl(mtl_obj):
+                    for key, value in mtl_obj.items():
+                        if isinstance(value, dict):
+                            for pair in iterate_mtl(value):
+                                yield (key, *pair)
+                        else:
+                            yield (key, value)
+
+                for pair in iterate_mtl(mtl):
+                    if "COLLECTION_NUMBER" in pair:
+                        COLLECTION_NUMBER = pair[pair.index("COLLECTION_NUMBER") + 1]
+                if "COLLECTION_NUMBER" not in locals():
+                    raise AttributeError("Cannot find COLLECTION_NUMBER in MTL metadata file")

```

Comment:
I would prefer doing it like this instead of checking in `locals()`:
```suggestion
                COLLECTION_NUMBER = next((pair[pair.index("COLLECTION_NUMBER") + 1] for pair in iterate_mtl() if "COLLECTION_NUMBER" in pair), None)
                if not "COLLECTION_NUMBER":
                    raise AttributeError("Cannot find COLLECTION_NUMBER in MTL metadata file")

```

Reply:
Good catch, thanks!